typescript/11-robot-name/robot-name.ts

```diff
@@ -0,0 +1,52 @@
+export class Robot {
+  private static usedNames: Set<string> = new Set();
+  private _name: string | null = null;
+
+  constructor() {}
+
+  private static generateRandomName(): string {
+    const letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    const randomLetters = (qty: number): string => {
+      let str = "";
+      while (str.length < qty) {
+        str += letters[Math.floor(Math.random() * letters.length)];
+      }
+      return str;
+    }
+
+    const randomDigits = (qty: number): string => {
+      let num = "";
+      while (num.length < qty) {
+        num +=  Math.floor(Math.random() * 10).toString();
+      }
+      return num;
+    }
+
+    return `${randomLetters(2)}${randomDigits(3)}`;
+  }
+
+  private static generateUniqueName(): string {
+    let name: string;
+    do {
+      name = this.generateRandomName();
+    } while (this.usedNames.has(name));
```

Comment:
Since the tests include generating `TOTAL_NUMBER_OF_NAMES` of names, this approach won't work.
As the number of generated names increase, so will the number of collisions, leading to a randomized, very high execution time.
Specifically, calling this `TOTAL_NUMBER_OF_NAMES` times will lead to a runtime of `O(n^2)`, randomized

Reply:
What's the difference between this approach and a previous one?